src/test/java/com/oracle/nosql/spring/data/test/composite/TestTableCreation.java

```diff
@@ -239,6 +239,249 @@ public void testCompositeEntityKvJsonField() {
         }
     }
 
+    @Test
+    public void testTableDDLMismatchOrderOfShardKeys() {
+        Class<?> domainClass = CompositeEntityWithNoKeys.class;
+        TableRequest tableRequest = new TableRequest();
+        String ddl = String.format("CREATE TABLE %s (ID2 STRING, ID1 STRING, " +
+                        "KV_JSON_ JSON, PRIMARY KEY(ID2,ID1))",
+                domainClass.getSimpleName());
+        tableRequest.setStatement(ddl);
+        tableRequest.setTableLimits(new TableLimits(100,100,1));
+        try {
+            template.dropTableIfExists(domainClass.getSimpleName());
+            template.getNosqlClient().doTableRequest(tableRequest, 10000, 2000);
+            template.createTableIfNotExists(template.getNosqlEntityInformation(domainClass));
+            fail("Expecting IllegalArgumentException but didn't get");
+        } catch (IllegalArgumentException ignored) {
+            template.dropTableIfExists(domainClass.getSimpleName());
```

Comment:
Here test should check the message of the exception.

Reply:
Added